lib/linter/linter.js

```diff
@@ -1326,19 +1326,38 @@ class Linter {
      */
     constructor({ cwd, configType = "flat", flags = [] } = {}) {
 
+        const processedFlags = [];
+
         flags.forEach(flag => {
             if (inactiveFlags.has(flag)) {
-                throw new Error(`The flag '${flag}' is inactive: ${inactiveFlags.get(flag)}`);
+                const { description, replacedBy } = inactiveFlags.get(flag);
+
+                if (typeof replacedBy === "undefined") {
+                    throw new Error(`The flag '${flag}' is inactive: ${description}`);
+                }
+
+                if (typeof replacedBy === "string") {
+                    processedFlags.push(replacedBy);
+                }
+
+                globalThis.process?.emitWarning?.(
+                    `The flag '${flag}' is inactive: ${description}`,
+                    `ESLintInactiveFlag_${flag}`
+                );
```

Comment:
I wasn't sure if `replacedBy` should be automatically included in the warning message (it isn't) or if we'll update the flag description to mention the new stable flag there.

Reply:
I think it would be a good idea to include the new flag name, something like:

> The flag 'unstable_foo' has been renamed `v10_foo` to reflect its stabilization. Please use `v10_foo` instead.

Reply:
Updated in https://github.com/eslint/eslint/pull/19386/commits/05fc5a4467ab91fc88f585c34f56e23bc5afd247